docs/v3/core/index.rst

```diff
@@ -1598,38 +1508,162 @@ Storage transformers may be stacked to combine different functionalities:
+array       chunk key encoding      :ref:`chunk-key-encoding <array-metadata-chunk-key-encoding>`      :ref:`chunk-key-encoding-list`
+array       codecs                  :ref:`codecs <array-metadata-codecs>`                              :ref:`codecs-list`
+array       storage transformer     :ref:`storage-transformers <array-metadata-storage-transformers>`  :ref:`storage-transformers-list`
+=========== ======================= ================================================================== ================================
+
+New extension points may be proposed to the Zarr community through the ZEP
+process. See `ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ for more information.
+
+.. _extension-definition:
+
+Extension definition
+--------------------
+
+In `metadata documents`_, extensions can be encoded either as objects or as
+short-hand names. If using an objection definition, the member ``name``
+MUST be a plain string which conforms to :ref:`extension name <extension-naming>`.
+Optionally, the member ``configuration`` MAY be present but if so MUST be
+an object.
+
+For example::
+
+    {
+        "name": "<name>",        # "raw name" or URL-based name
+        "configuration": { ... } # optional
+    }
+
+Instead of extension objects, short-hand names MAY be used if no
+configuration metadata is required. They are equivalent to extension
+objects with just a `name` key.
+
+If such an object is present, the field `must_understand` is implicitly set to
+`True` and an object MAY explicitly set `must_understand=False` if
+implementations can ignore its presence.
+
+An implementation MUST fail to open Zarr groups or arrays if any
+metadata fields are present which (a) the
+implementation does not recognize and (b) are not explicitly
+set to ``"must_understand": false``.
+
+`must_understand=False` is not supported for the following extension points:
+data type, chunk grid, and chunk key encoding.
+
+.. _extension-naming:
+
+Extension naming
+----------------
+
+The `name` field of an extension can take two forms: **raw names** and **URL-based names**.
+
+.. _extension-naming-raw-names:
+
+Raw names
+^^^^^^^^^
+
+Raw names are centrally registered names which can be used without prefix.
+
+Raw names MUST be assigned within a central repository.
+Raw names are unique and immutable.
+Raw names MUST be composed of lower case letters a-z, numerals 0-9, underscores, dashes, and dots.
+Raw name assignment is managed through the `zarr-extensions`_
+Github repository, where extensions and their specification are listed.
+The Zarr Steering Council or by delegation a
+maintainer team reserves the right to refuse name assignment at its own
+discretion.
+
+- **Example:** ``zstd``
+- **Acceptd regex:** ``^[a-z0-9-_.]+$``
+
+.. _extension-naming-url-based-names:
+
+URL-based names
+^^^^^^^^^^^^^^^
+
+URL-based names delegate name registration to the domain name system (DNS).
```

Comment:
That is not entirely right, because paths are valid. In fact, I could imagine many URL-based names would be some https://github.com/... URL.

Reply:
```suggestion
URL-based names delegate name registration to URLs on the internet.
```

Reply:
Very true. I'll try to say it more explicitly.